internal/report/source.go

```diff
@@ -132,11 +132,14 @@ func printWebSource(w io.Writer, rpt *Report, obj plugin.ObjTool) error {
 // sourcePrinter holds state needed for generating source+asm HTML listing.
 type sourcePrinter struct {
 	reader     *sourceReader
+	mappings   mappingList
+	synth      synthCode
```

Comment:
I wonder if the code would be simpler if we restrict the case of absent mappings to zero address, rather than any address, for now. I believe I'm not aware of practical cases where the address is present and the mapping is missing.